src/main/java/registrar/Register.java

```diff
@@ -0,0 +1,8 @@
+package registrar;
+
+public class Register {
+    private RegisteredRepository repository;
+    public void add (String email, String name) {
+        repository.save(new Registered(email, name));
```

Comment:
Im probably missing something to instantiate repository (RegisteredRepository is abstract)

Reply:
You are missing the annotation @autowired on top of it (it will initialize it)

Reply:
> You are missing the annotation @Autowired on top of it (it will initialize it)
I'm not sure how this Autowired works. My code started to work when I added the constructor instead of using the Autowired.
I'm merging this one, but I will change few things later so I'll take a deeper look at this annotation.